### PR TITLE
Port tests network refactoring.

### PR DESCRIPTION
**Pull request by AurelienFT**
- [x] ask_for_blocks_scenarios.rs
- [x] ban_nodes_scenarios.rs
- [x] cache_scenarios.rs
- [x] endorsements_scenarios.rs
- [x] in_block_operation_scenarios.rs
- [x] operation_scenarios.rs
- [x] scenarios.rs

### CODE REVIEW

File: massa-protocol-worker-2/src/handlers/block_handler/propagation.rs [hunk 1]
```diff
@@ -39,18 +43,28 @@ impl PropagationThread {
...
-                                    })
-                                    .unwrap()
+                                let block = {
+                                    let blocks = storage.read_blocks();
+                                    blocks.get(&block_id).cloned()
+                                };
+                                if let Some(block) = block {
+                                    self.storage.store_block(block.clone());
+                                    self.saved_blocks.push_back(block.id);
+                                    if self.saved_blocks.len()
```
Comment (modship) [inline massa-protocol-worker-2/src/handlers/block_handler/propagation.rs:53]: maybe check len() + 1 before insert on it and not after + rollback ?

File: massa-protocol-worker-2/src/tests/ban_nodes_scenarios.rs [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+                .send_from_peer(
+                    &node_a_peer_id,
+                    Message::Block(Box::new(BlockMessage::BlockHeader(
+                        block.content.header.clone(),
+                    ))),
+                )
+                .unwrap();
+
+            std::thread::sleep(std::time::Duration::from_millis(1000));
+            //4. Check that node is banned.
```
Comment (modship) [inline massa-protocol-worker-2/src/tests/ban_nodes_scenarios.rs:59]: in fact that check only if connection with remote peer is closed

File: massa-protocol-worker-2/src/handlers/block_handler/propagation.rs (hunk 1 above)
Comment (AurelienFT) [reply massa-protocol-worker-2/src/handlers/block_handler/propagation.rs:53]: I don't rollback I remove the oldest one that is at front but keep the one I just inserted which is at the back.

Comment (modship) [reply massa-protocol-worker-2/src/handlers/block_handler/propagation.rs:53]: oh yeah sorry !

File: massa-protocol-worker-2/src/tests/ban_nodes_scenarios.rs (hunk 2 above)
Comment (modship) [reply massa-protocol-worker-2/src/tests/ban_nodes_scenarios.rs:59]: yes no problem, just update the comment like `Check that node connection is closed (node should be banned)`

Comment (AurelienFT) [reply massa-protocol-worker-2/src/tests/ban_nodes_scenarios.rs:59]: ok

**Review by modship — APPROVED** [review]

**Pull request merged**